### PR TITLE
RHPAM-3557 :Slow performance opening assets in projects with large commit history

### PR DESCRIPTION
**Pull request by RishiRajAnand**
- some code level optimization to help with this

**Thank you for submitting this pull request**

**RHPAM-3557**: https://issues.redhat.com/browse/RHPAM-3557

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitBasicAttributeView.java [hunk 1]
```diff
@@ -71,24 +71,30 @@ private BasicFileAttributes buildAttrs(final JGitFileSystem fs,
...
-                    records.add(commit.getAuthorIdent().getWhen());
-                }
+                List<RevCommit> rec  = history.getCommits();
+                result[0]  = rec.get(0).getAuthorIdent().getWhen().getTime();
+                result[1]  = rec.get(rec.size() - 1).getAuthorIdent().getWhen().getTime();
+
+//                for (final RevCommit commit : history.getCommits()) {
+//                    records.add(commit.getAuthorIdent().getWhen());
+//                }
             } catch (Exception e) {
```
Comment (martinweiler) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitBasicAttributeView.java:88]: @RishiRajAnand I tested this change with a fairly large niogit repo, and could not see a performance improvement. Upon further testing, I realized this costly lookup of the jgit history is happening too frequently at this location, so I changed this to a lazy approach whenever one of the lastModifiedTime() or creationTime() methods are called, see PR#1132 for details. With this change, performance is back to where it was before the RHPAM-2865 change. Please let me know what you think.

Comment (adrielparedes) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitBasicAttributeView.java:88]: @martinweiler I commented a few Stuff in PR#1132 that I think we need to change (very small stuff) and if the performance is better we should ask QE to review and confirm. Good job!

Comment (martinweiler) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitBasicAttributeView.java:88]: @adrielparedes thanks, I have provided the numbers from my local testing in the other PR.

**Comment by RishiRajAnand** [comment]
@adrielparedes I have refactored the code, please let me know if need anything here

**Comment by adrielparedes** [comment]
@RishiRajAnand @martinweiler I will approve, let's see what QE says, good job guys

**Comment by tomasdavidorg** [comment]
Hi @adrielparedes  @RishiRajAnand , I tested this PR and I see that opening asset with long commit history is faster. I did some testing on my local machine:
Avg. Opening 1 asset with 1000 commits took:
- 7.10: 8s
- This PR: 1s

Avg. Opening 1 asset with 10000 commits took:
- 7.10: 23s
- This PR 8s

Do I understand correctly that this PR is the same as  #1132 with some refactoring? So we can close #1132?

**Review by tomasdavidorg — APPROVED** [review]

**Comment by adrielparedes** [comment]
thanks @tomasdavidorg !

**Review by adrielparedes — APPROVED** [review]

**Comment by adrielparedes** [comment]
@RishiRajAnand @martinweiler are 1131 and 1132 related? I approved both but we need confirmation so @tomasdavidorg can test it or not. I already approved

**Comment by martinweiler** [comment]
I closed 1132 now as we have everything required in this PR. I hope it can be merged now. Thanks!

**Comment by martinweiler** [comment]
@adrielparedes where are we on getting this fix merged? @tomasdavidorg has posted the test results for an asset with a large commit history above. Is anything else required? Sorry for pushing, but we have an urgent request for a one-off patch. Thanks!

**Comment by ederign** [comment]
@mareknovotny  as soon as we have a green build, we can merge it.

**Comment by adrielparedes** [comment]
jenkins run fdb

**Comment by adrielparedes** [comment]
@ederign the failing test is not related: ready to merge

**Pull request merged**